### PR TITLE
Add FileWebRequest/Response to corefx

### PR DESCRIPTION
**Pull request by stephentoub**
The semantics of these types are a bit strange in places (e.g. allowing a request stream and a response stream for the same file, but using a null stream for the response in that case), but I didn't mess with them.

cc: @davidsh, @cipop, @ericeil, @karelz, @geoffkizer

### CODE REVIEW

File: src/System.Net.Requests/tests/FileWebRequestTest.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                    Assert.Equal(0, s.Length);
+                }
+            }
+            finally
+            {
+                File.Delete(path);
+            }
+        }
+
+
```
Comment (davidsh) [inline src/System.Net.Requests/tests/FileWebRequestTest.cs]: nit: remove duplicated blank line.

Comment (stephentoub) [inline src/System.Net.Requests/tests/FileWebRequestTest.cs]: Will fix

File: src/System.Net.Requests/src/System/Net/FileWebRequest.cs [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+        internal void UnblockReader()
+        {
+            lock (this) { _blockReaderUntilRequestStreamDisposed?.Set(); }
+            _writing = false;
+        }
+
+        public override bool UseDefaultCredentials
+        {
+            get { throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException); }
+            set { throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException); }
```
Comment (davidsh) [inline src/System.Net.Requests/src/System/Net/FileWebRequest.cs]: Is this the behavior from Desktop?  It's rare we would throw an exception on a property getter.

Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/FileWebRequest.cs]: Yup: http://referencesource.microsoft.com/#System/net/System/Net/filewebrequest.cs,503

**Comment by stephentoub** [comment]
> Hopefully you ported as many tests from Desktop as well.

I did.

**Comment by stephentoub** [comment]
@dotnet-bot Test this please (Jenkins hasn't restarted after an updated commit)

**Comment by stephentoub** [comment]
@dotnet-bot Test this please (Jenkins hasn't restarted after an updated commit)

**Comment by stephentoub** [comment]
@dotnet-bot Test Innerloop CentOS7.1 Release Build and Test please (known seg fault)
@dotnet-bot Test  Innerloop Ubuntu14.04 Release Build and Test please (known OS bug)

**Pull request merged**